src/main/java/duke/main/Utils.java

```diff
@@ -0,0 +1,192 @@
+        File f = new File("list.txt"); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            loadTask(s.nextLine());
+        }
+    }
+
+    public static void loadTask(String line){
+        String[] arr = line.split("\t");
+        switch(arr[0]){
+        case ("T"):
+            Todo current1 = new Todo(arr[2]);
+            if (arr[1].equals("true")) {
+                current1.setDone();
+            }
+            Task.addTask(current1);
+            break;
+
+        case ("D"):
+            Deadline current2 = new Deadline(arr[2], arr[3]);
+            if (arr[1].equals("true")) {
+                current2.setDone();
+            }
+            Task.addTask(current2);
+            break;
+
+        case ("E"):
+            Event current3 = new Event(arr[2], arr[3]);
+            if (arr[1].equals("true")) {
+                current3.setDone();
+            }
+            Task.addTask(current3);
+            break;
+
+        }
+    }
+
+
+    public static int commandHandler(String line) throws InvalidCommandExceptions,InvalidParameterLengthExceptions,
+            DeadlineParameterExceptions, EventParameterExceptions, InvalidIndexExceptions, IOException {
+
+        String[] arr;
+        arr = line.split(" ");
+        if (validateCommand(arr[0]) == 0){
+            throw new InvalidCommandExceptions();
+        }
+        switch (arr[0]) {
+
+        case ("bye"):
+            writeToFile();
+            return -1;
+
+        case ("list"):
+            System.out.println("Here are the tasks in your list:");
+            Task.printList();
+            break;
+        case ("done"):
+            Task.setDone(Integer.parseInt(line.split(" ")[1]) - 1);
+            break;
+        case ("delete"):
+            Task.deleteTask(Integer.parseInt(line.split(" ")[1]) - 1);
+            break;
+        default:
+            int indexOfSpace = line.indexOf(" ");
+            if (line.split(" ").length < 2){
+                throw new InvalidParameterLengthExceptions();
+            }
+            String type = line.substring(0, indexOfSpace);
+
+            switch (type) {
+            case ("todo"):
+                System.out.println("Got it. I've added this task: ");
+                Task.addTask(new Todo(line.substring(indexOfSpace + 1)));
+                System.out.println("  [T][\u2718] " + line.substring(indexOfSpace + 1));
+                break;
+
+            case ("deadline"):
+                int indexOfSlash = validateParameter(line);
+                if (indexOfSlash == -1){
+                    throw new DeadlineParameterExceptions();
+                }
+                System.out.println("Got it. I've added this task: ");
+                String item = line.substring(indexOfSpace + 1, indexOfSlash - 1);
+                String extra = line.substring(indexOfSlash + 4);
```

Comment:
maybe we should have some code comments here explaining the logic behind +1,-1,+4?